client.go

```diff
@@ -38,6 +40,10 @@ type ClientInterface interface {
 	GetVersion() (resp *Version, err error)
 	Health() (*Health, error)
 	IsHealthy() bool
+	GetTask(taskID int64) (resp *Task, err error)
+	GetTasks() (resp *ResultTask, err error)
+	WaitForTask(ctx context.Context, interval time.Duration, taskID *Task) (*Task, error)
+	DefaultWaitForTask(taskID *Task) (*Task, error)
```

Comment:
What is `defaultWaitForTask` ?

I see that it is when you provide the default values. Is this the naming convention in GO ?

Reply:
It is the `WaitForTask` with default values. I don't think this naming is a specific convention in go. I can rename it if you have a better naming 🤩

Reply:
in `waitForTask` you can add a structure the following way
```
type waitParams struct{
   interval: int,
   taskId: int
}
```

provide this as a parameter and in case of null use the default value?

```go
function waitForTask(taskId: int, params WaitParams) {
   var interval = 500 // default value
   if (params.interval != 0) {
     interval = params.interval
  }
  // etc ...
}

```

would something like that be possible?

Reply:
Yes, but you had to specify the last param at `nil` like:
```go
	client. waitForTask(0, nil)
```
There is no optional parameters in `golang` but I can use variadic args and do something like:
```go
func (c *Client) WaitForTask(task *Task, options ...waitParams) (*Task, error) {
	if (options == nil) {
		ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second*5)
		defer cancelFunc()
		options[0].Context = ctx
		options[0].Interval = time.Millisecond*50
	}
... do waitForTask ...
}
```
And I can use it like:
```
	client. waitForTask(0)
	// OR
	client. waitForTask(0, &waitParams{ctx, time.Millisecond*10})
```
I'm not sure which one is the most elegant because in the second option I get the `waitParams` as an array.
What do you think?